examples/EDMF_data/uq_for_edmf.jl

```diff
@@ -0,0 +1,246 @@
+#include(joinpath(@__DIR__, "..", "ci", "linkfig.jl"))
+PLOT_FLAG = true
+
+# Import modules
+using Distributions  # probability distributions and associated functions
+using LinearAlgebra
+using Plots
+using Random
+using JLD2
+using NCDatasets
+using StatsBase
+using Dates
+
+# CES 
+using CalibrateEmulateSample.Emulators
+using CalibrateEmulateSample.MarkovChainMonteCarlo
+using CalibrateEmulateSample.ParameterDistributions
+using CalibrateEmulateSample.DataContainers
+using CalibrateEmulateSample.EnsembleKalmanProcesses
+using CalibrateEmulateSample.Utilities
+
+rng_seed = 42424242
+Random.seed!(rng_seed)
+
+
+function main()
+
+    # 2-parameter calibration exp
+    exp_name = "ent-det-calibration"
+
+    # 5-parameter calibration exp
+    #exp_name = "ent-det-tked-tkee-stab-calibration"
+
+
+    # Output figure save directory
+    figure_save_directory = joinpath(@__DIR__, "output", exp_name, string(Dates.today()))
+    data_save_directory = joinpath(@__DIR__, "output", exp_name, string(Dates.today()))
+    if !isdir(figure_save_directory)
+        mkpath(figure_save_directory)
+    end
+    if !isdir(data_save_directory)
+        mkpath(data_save_directory)
+    end
+    println("experiment running in: ", @__DIR__)
+    println("data saved in: ", figure_save_directory)
+    println("figures saved in: ", data_save_directory)
+
+    #############
+    # Pipeline: #
+    #############
+
+    # [1. ] Obtain scenario information (here all loaded from file)
+    # [a.]  Obtain truth sample(s),
+    # [b.]  Obtain observation and model-error noise 
+    # [2. ] Obtain calibration information  (here using saved input-output data, from EKI experiements)
+    # [a.]  Obtain calibration priors
+    # [b.]  Obtain calibration data
+    # [3. ] Build Emulator from calibration data
+    # [4. ] Run Emulator-based MCMC to obtain joint parameter distribution
+
+    ###################################################
+    # [1. & 2. ] Obtain scenario and calibration data #
+    ###################################################
+
+    exp_dir = joinpath(@__DIR__, exp_name)
+    if !isdir(exp_dir)
+        LoadError(
+            "experiment data directory \"" *
+            exp_dir *
+            "/\" not found. Please unzip the compressed file \"" *
+            exp_name *
+            ".zip\" and retry.",
+        )
+    end
+    data_filepath = joinpath(exp_dir, "Diagnostics.nc")
+    if !isfile(data_filepath)
+        LoadError("experiment data file \"Diagnostics.nc\" not found in directory \"" * exp_dir * "/\"")
+    else
+        y_truth = Array(NCDataset(data_filepath).group["reference"]["y_full"]) #ndata
+        truth_cov = Array(NCDataset(data_filepath).group["reference"]["Gamma_full"]) #ndata x ndata
+        # Option (i) get data from NCDataset else get from jld2 files.
+        output_mat = Array(NCDataset(data_filepath).group["particle_diags"]["g_full"]) #nens x ndata x nit
```

Comment:
Remove last iteration (it has NCDataset fill values)

Reply:
You can also check whether a value is bad by using:

```
isnan_inf_or_filled(x) = isnan(x) || isinf(x) || x ≈ NC.fillvalue(typeof(x))
```

we use it in TC here https://github.com/CliMA/TurbulenceConvection.jl/blob/main/integration_tests/driver.jl#L76

Reply:
done